thinc/layers/with_flatten.py

```diff
@@ -2,49 +2,64 @@
 
 from ..model import Model
 from ..config import registry
-from ..types import ListXd
 
 
+InItemT = TypeVar("InItemT")
+OutItemT = TypeVar("OutItemT")
 ItemT = TypeVar("ItemT")
-InT = Sequence[Sequence[ItemT]]
-OutT = TypeVar("OutT", bound=ListXd)
+
+NestedT = List[List[ItemT]]
+FlatT = List[ItemT]
 
 
 @registry.layers("with_flatten.v1")
```

Comment:
I doubt anyone was using this, but to be on the safe side I think this should be `with_flatten.v2`.

Reply:
I'll do that, but I think that we could then have the docs only describe v2 since the documentation was not accurate for v1 anyway?

Reply:
Added back, also tried to fix the typing of the old layer as much as possible, since they didn't reflect what was happening in the old layer.

Reply:
I would be tempted to just leave it completely alone, since it doesn't sound like anyone would want to use it?

Reply:
Hmm, I don't really know what do about the docs now, though, since `with_flatten_v2` looks kind of awful.

Reply:
Do you mean the name? We already have `v2`s of the sparse linear and softmax layers.

Reply:
I guess there's no good way around it.